app/page.tsx

```diff
@@ -1,15 +1,75 @@
+"use client";
+import useSWR from "swr";
 import Hero from "./components/Hero-section/Hero";
 import MoviesSection from "./components/MoviesSection/MoviesSection";
-import { movies, tabs } from "./mockdata";
+import { useState } from "react";
+import { apiBaseUrl, apiKey, fetcher, imgUrl } from "./api";
+import { popupMenuLinks } from "./mockdata";
 export default function Home() {
+  const [timeWindow, setTimeWindow] = useState<"day" | "week">("day");
+  const { data: popularData, error: popularError } = useSWR(
+    `${apiBaseUrl}/3/movie/popular?api_key=${apiKey}`,
+    fetcher
+  );
+  const { data: trendingData, error: trendingError } = useSWR(
+    `${apiBaseUrl}/3/trending/movie/${timeWindow}?api_key=945747413f8ed82865430ae895bdf874`,
+    fetcher
+  );
+
+  if (!popularData && !popularError) {
+    return <div>...Loading...</div>;
+  }
+
+  if (popularError) {
+    return <div>Error: {popularError.message || "An error occurred"}</div>;
+  }
+  const popularMovies = popularData.results.map((movie: any) => {
+    return {
+      movieImage: {
+        imgSrc: `${imgUrl}${movie.poster_path}`,
+        altText: "Image Not Found",
+        url: "/",
+      },
+      title: movie.title,
+      date: movie.release_date,
+      rating: Math.floor(movie.vote_average * 10),
+      popupMenuLinks: popupMenuLinks,
+    };
+  });
+
+  if (!trendingData && !trendingError) {
+    return <div>...Loading...</div>;
+  }
+
+  if (trendingError) {
+    return <div>Error: {trendingError.message || "An error occurred"}</div>;
+  }
+  const trendingMovies = trendingData.results.map((movie: any) => {
+    return {
+      movieImage: {
+        imgSrc: `${imgUrl}${movie.poster_path}`,
+        altText: "Image Not Found",
+        url: "/",
+      },
+      title: movie.title,
+      date: movie.release_date,
+      rating: Math.floor(movie.vote_average * 10),
+      popupMenuLinks: popupMenuLinks,
+    };
+  });
+
+  let tabs = [
+    { tabName: "Today", onClick: () => setTimeWindow("day") },
+    { tabName: "This Week", onClick: () => setTimeWindow("week") },
+  ];
```

Comment:
Since we are doing the same thing twice, can we create a fumction to do this action and call the function twice?
Ex. 
const getFormattedMovies = (movies)=> movies.map((movie: any) => {
    return {
      movieImage: {
        imgSrc: `${imgUrl}${movie.poster_path}`,
        altText: "Image Not Found",
        url: "/",
      },
      title: movie.title,
      date: movie.release_date,
      rating: Math.floor(movie.vote_average * 10),
      popupMenuLinks: popupMenuLinks,
    };
  });


then 
const trendingMovies = getFormattedMovies(trendingData.results).
const popularMovies = getFormattedMovies(popularData.results).